ledger-core/v2/network/messagesender/adapter/init.go

```diff
@@ -12,12 +12,19 @@ import (
 	"github.com/insolar/assured-ledger/ledger-core/v2/network/messagesender"
 )
 
-type MessageSender struct {
+//go:generate minimock -i github.com/insolar/assured-ledger/ledger-core/v2/network/messagesender/adapter.MessageSender -o ./ -s _mock.go -g
+type MessageSender interface {
+	PrepareSync(ctx smachine.ExecutionContext, fn func(svc messagesender.Service)) smachine.SyncCallRequester
+	PrepareAsync(ctx smachine.ExecutionContext, fn func(svc messagesender.Service) smachine.AsyncResultFunc) smachine.AsyncCallRequester
+	PrepareNotify(ctx smachine.ExecutionContext, fn func(svc messagesender.Service)) smachine.NotifyRequester
+}
+
+type ParallelMessageSender struct {
```

Comment:
Why parallel?

Reply:
Because we have only one implementation and i try to decide witch name better shows it destination. This implementation use `smachine.StartChannelWorkerParallelCalls` and i decide that Parallel is important property of this implementation